i686-pc-windows-gnu fails to link on appveyor

```
error: linking with `i686-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "i686-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-Wl,--large-address-aware" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\crt2.o" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.0.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.1.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.10.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.11.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.12.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.13.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.14.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.15.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.2.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.3.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.4.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.5.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.6.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.7.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.8.rcgu.o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.build_script_build.aj5y0a64-cgu.9.rcgu.o" "-o" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.exe" "C:\\projects\\winapi-rs\\target\\debug\\build\\winapi-i686-pc-windows-gnu-443effcf81014333\\build_script_build-443effcf81014333.r46olty88oc842v.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "C:\\projects\\winapi-rs\\target\\debug\\deps" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-Wl,--start-group" "-Wl,-Bstatic" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libstd-7dc27a6f3096fff5.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libpanic_unwind-b9fb1be3dc76db82.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libhashbrown-52ece7fac3462bf3.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_std_workspace_alloc-92c58a1a3a774a1f.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libbacktrace-d14a966be767447d.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libbacktrace_sys-ceb056ead5f4dd46.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_demangle-5aaa5927600b79eb.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libunwind-c470ca234bf4591f.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcfg_if-c0a586ae7f2fa67b.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\liblibc-81876339e21271aa.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\liballoc-1b8d5d0b1b608063.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\librustc_std_workspace_core-8f97c41df79c34b0.rlib" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcore-657b4379f124771d.rlib" "-Wl,--end-group" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcompiler_builtins-c7e55169dc2dbc63.rlib" "-Wl,-Bdynamic" "-ladvapi32" "-lws2_32" "-luserenv" "-lmingwex" "-lmingw32" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "-lgcc" "-lgcc_eh" "-lpthread" "-lkernel32" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-i686-pc-windows-gnu\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsend.o"
  = note: C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0x631): undefined reference to `_imp___ultoa'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0xbf7): undefined reference to `_setjmp3'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0xc8d): undefined reference to `_imp___endthreadex'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0xda8): undefined reference to `printf'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0xdd6): undefined reference to `printf'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0x14e2): undefined reference to `longjmp'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0x152e): undefined reference to `_imp___endthreadex'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0x1f47): undefined reference to `_imp___beginthreadex'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-thread.o):thread.c:(.text+0x2484): undefined reference to `strdup'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-rwlock.o):rwlock.c:(.text+0x44e): undefined reference to `printf'
          C:\Users\appveyor\.rustup\toolchains\nightly-i686-pc-windows-gnu\lib\rustlib\i686-pc-windows-gnu\lib/libpthread.a(libwinpthread_la-rwlock.o):rwlock.c:(.text+0x480): undefined reference to `printf'
          

error: aborting due to previous error
```
Using `image: Visual Studio 2017`. No environment changes other than installing Rust itself.

--- a/src/librustc_target/spec/windows_base.rs
+++ b/src/librustc_target/spec/windows_base.rs
@@ -60,6 +60,8 @@ pub fn opts() -> TargetOptions {
             "-lgcc".to_string(),
             "-lgcc_eh".to_string(),
             "-lpthread".to_string(),
+            // libpthread depends on libmsvcrt, so we need to link it *again*.
+            "-lmsvcrt".to_string(),
             "-lkernel32".to_string(),
         ],
     );

